app/backend/lib/awsCommon.ts

```diff
@@ -1,31 +1,39 @@
-import https from 'https'; 
+import http from 'https';
+import { NodeHttpHandler } from '@aws-sdk/node-http-handler';
 import {
   fromTemporaryCredentials,
   fromEnv,
 } from '@aws-sdk/credential-providers';
-
 import config from '../../config';
+import CustomHttpsAgent from './CustomHttpsAgent';
 
-const AWS_S3_REGION = config.get('AWS_S3_REGION'); 
-const AWS_ROLE_ARN = config.get('AWS_ROLE_ARN'); 
+const AWS_S3_REGION = config.get('AWS_S3_REGION');
+const AWS_ROLE_ARN = config.get('AWS_ROLE_ARN');
 
-const agent = new https.Agent({
+const httpsAgent = new CustomHttpsAgent();
+const httpAgent = new http.Agent({
   maxSockets: 300,
-  keepAlive: true
- });
+  keepAlive: true,
+});
+
+const nodeHandler = new NodeHttpHandler({
+  httpAgent,
+  httpsAgent,
```

Comment:
Here we can set the connectionTimeout to 30 seconds rather than the awsConfig object